### PR TITLE
Mapper changes due to the BlockDisambiguate pass.

### PR DESCRIPTION
**Pull request by vext01**
The mapper can now unambiguously assume that consecutive repeated IR
blocks in a trace correspond to "internal" control flow between multiple
machine blocks of a single LLVM IR block.

This is the companion to https://github.com/ykjit/ykllvm/pull/11 which should be merged first.

### CODE REVIEW

File: yktrace/src/hwt/mapper.rs [hunk 1]
```diff
@@ -130,11 +130,11 @@ impl HWTMapper {
...
                     if let Some(irblock) = irblock {
-                        // The `BlockDisambiguate` pass in ykllvm ensures that no LLVM IR block
-                        // ever branches straight back to itself, so if we see the same IR block
-                        // more than once consecutively in a trace, then we know that the IR block
-                        // has code-genned to multiple machine blocks. In this case we should only
-                        // push the IR block once.
+                        // The `BlockDisambiguate` pass in ykllvm ensures that no high-level LLVM
+                        // IR block ever branches straight back to itself, so if we see the same
+                        // high-level block more than once consecutively in a trace, then we know
+                        // that the IR block has code-genned to multiple machine blocks, and that
```
[the remark this answers is not in the document]
Comment (vext01) [reply yktrace/src/hwt/mapper.rs]: bf2db97 dodges this wording.

**Comment by ltratt** [comment]
Please squash.

**Comment by vext01** [comment]
Squashed and synced. The ykllvm PR can now be merged.

**Pull request merged**